Sources/AsyncKit/ConnectionPool/EventLoopConnectionPool.swift

```diff
@@ -261,34 +278,55 @@ public final class EventLoopConnectionPool<Source> where Source: ConnectionPoolS
             self.eventLoop.execute { self._releaseConnection0(connection, logger: logger) }
         }
     }
-    
+
     private func _releaseConnection0(_ connection: Source.Connection, logger: Logger) {
         self.eventLoop.assertInEventLoop()
-        
+
         // If the pool has shut down, just close the connection and return
         guard !self.didShutdown else {
             if !connection.isClosed {
                 _ = connection.close()
             }
             return
         }
-        
+
         // Push the connection onto the end of the available list so it's the first one to get used
         // on the next request.
-        logger.trace("Releasing pool connection on \(self.eventLoop.description), \(self.available.count + (connection.isClosed ? 0 : 1)) connnections available.")
-        self.available.append(connection)
-        
+        logger.trace("Releasing pool connection on \(self.eventLoop.description), \(self.available.count + (connection.isClosed ? 0 : 1)) connections available.")
+
+        self.available.append((connection: connection, lastUse: .now()))
+        if self.available.count == 1, let pruningInterval = self.pruningInterval, self.pruningTask == nil { // just added our very first connection
+            self.pruningTask = self.eventLoop.scheduleRepeatedTask(
+                initialDelay: pruningInterval,
+                delay: pruningInterval
+            ) { [unowned self] _ in
+                var prunedConnections: Set<ObjectIdentifier> = []
+
+                for conninfo in self.available where !conninfo.connection.isClosed {
+                    if conninfo.lastUse + self.maxIdleTimeBeforePruning < .now() {
```

Comment:
nit, maybe a `conninfo.lastUse < .now() - self.maxIdleTimeBeforePruning` where `.now() - self.maxIdleTimeBeforePruning` is calculated out of the loop and assigned to a variable?

Not sure if Swift is smart enough to optimize this?